aclocal.m4 contains code that no longer compiles: missing headers and missing
return types.

https://bugs.gentoo.org/875458
https://bugs.gentoo.org/921487
https://bugs.gentoo.org/926491

--- a/autoconf_local/aclocal.m4
+++ b/autoconf_local/aclocal.m4
@@ -464,6 +464,7 @@ CFLAGS="$CFLAGS $ac_cv_ssp_flags"
 changequote([, ])dnl
 AC_TRY_LINK([
 #include <stdio.h>
+#include <string.h>
 #ifdef HAVE_STDLIB_H
 #	include <stdlib.h>
 #endif
@@ -1063,6 +1064,7 @@ AC_DEFUN(wi_USE_STATIC_LIBGCC, [
 				#
 				AC_TRY_LINK([
 #include <stdio.h>
+#include <string.h>
 #ifdef HAVE_STDLIB_H
 #	include <stdlib.h>
 #endif
@@ -3057,6 +3059,7 @@ AC_TRY_LINK([
 #include <sys/uio.h>
 #include <sys/socket.h>
 #include <sys/un.h>
+#include <string.h>
 ],[
 struct sockaddr_un uaddr;
 
@@ -3285,6 +3288,7 @@ AC_TRY_LINK([
 #ifdef HAVE_RESOLV_H
 #	include <resolv.h>
 #endif
+#include <string.h>
 ],[
 	/* function-body */
 	int len;
@@ -3363,6 +3367,7 @@ else
 #include <stdio.h>
 #include <stdlib.h>
 		 
+			int
 			main(int argc, char **argv)
 			{
 				/* Note:  don't actually call readline,
@@ -4193,7 +4198,7 @@ AC_MSG_CHECKING([types of arguments for select()])
 #include <sys/select.h>
 #endif
 #include <sys/socket.h>	/* MG: always incl */
-extern select ($ac_cv_func_select_arg1,$ac_cv_func_select_arg234,$ac_cv_func_select_arg234,$ac_cv_func_select_arg234,$ac_cv_func_select_arg5);],,dnl
+extern int select ($ac_cv_func_select_arg1,$ac_cv_func_select_arg234,$ac_cv_func_select_arg234,$ac_cv_func_select_arg234,$ac_cv_func_select_arg5);],,dnl
         [ac_not_found=no ; break 3],ac_not_found=yes)
       done
      done
@@ -5145,6 +5150,7 @@ do
 #ifdef HAVE_STDLIB_H
 #	include <stdlib.h>
 #endif
+#include <string.h>
 
 #ifdef HAVE_NCURSES_H
 #	define NCURSES_OPAQUE 0
@@ -5176,6 +5182,7 @@ do
 #endif
 
  
+int
 main(int argc, char **argv)
 {
 	/* Note:  don't actually call curses, since it may block;
@@ -6510,7 +6517,6 @@ char *malloc();
 #endif
 #endif			/* MG */
 
-int
 int  main(void)
 {
 	char *data, *data2, *data3;
@@ -6607,8 +6613,9 @@ else
 	AC_CACHE_CHECK(whether setvbuf arguments are reversed,
 	  ac_cv_func_setvbuf_reversed,
 	[AC_TRY_RUN([#include <stdio.h>
+	$include <stdlib.h>
 	/* If setvbuf has the reversed format, exit 0. */
-	main () {
+	int main (void) {
 	  /* This call has the arguments reversed.
 	     A reversed system may check and see that the address of main
 	     is not _IOLBF, _IONBF, or _IOFBF, and return nonzero.  */
